src/backends/torch/torchlib.cc

```diff
@@ -211,37 +219,48 @@ namespace dd
       torch::load(_classif, model._weights, _device);
     if (!model._proto.empty())
       {
-        _native = std::make_shared<CaffeToTorch>(model._proto);
+        _graph = std::make_shared<CaffeToTorch>(model._proto);
         if (!model._traced.empty())
-          torch::load(_native, model._traced, _device);
+          torch::load(_graph, model._traced, _device);
+      }
+    if (!model._native.empty())
+      {
+        std::shared_ptr<NativeModule> m;
+        torch::load(m, model._native);
+        _native = m;
       }
   }
 
   void TorchModule::eval()
   {
-    if (_native)
-      _native->eval();
+    if (_graph)
+      _graph->eval();
     if (_traced)
       _traced->eval();
     if (_classif)
       _classif->eval();
+    if (_native)
```

Comment:
Pointing the possible need of `else` statements here, may be inappropriate though, unsure.

Reply:
see first comment : this allows to chain nets for free, unused ATM